AGPs/AGP-1H4.md

```diff
@@ -0,0 +1,46 @@
+---
+AGP: tbd
+Title: Improved Deployment Processes and Testnet Support
+Author: @lkngtn (@1hive)
+Status: Stage II
+Track: Finance
+Created: 2019-11-12
+---
+
+# **AGP-TBD: Improved Deployment Processes and Testnet Support**
+
+## **Address of the transfer recipient**
+
+ `0x9213737fbb72d8f924fca55d508ae2c004413d64` Vault of the [dandelion project organization](https://mainnet.aragon.org/#/1hive)
+
+## **Amount of the transfer**
+
+15,000 DAI Due upon approval
+5,000 ANT Due upon successful delivery
+
+## **Number and frequency of transfers if recurring**
+
+2 (One upon approval, one upon successful delivery).
+For the purpose of budgeting, the full amount should be debited against this quarter's budget.
+
+## **Purpose of the transfer**
+
+Testnet deployment is an important part of the development process. However, it is currently difficult to setup an Aragon Development environment outside of the Ganache snapshot provided by the Aragon CLI. 1Hive has experienced this issue working on Apiary, where we need to have Aragon running in a local test environment with a real node. As well as when working on [Flora](https://devpost.com/software/flora-sidechain-discount-transactions-q8usv3)
+
+Additionally, there have been instances where the lack of support for Goerli, Ropsten, and Kovan testnets has produced friction between projects that would like to integrate Aragon into their protocols or dapps.
+
+To improve this we would like to improve deployment infrastructure and documentation in order to make it easy to deploy and maintain the Aragon environment on multiple blockchain networks.
+
+In exchange for the funding requested we will:
+* Make it easier to deploy the Aragon environment to a local chain in order to support development workflows that need access to a full node as opposed to a ganache snapshot.
+* We will improve and automate aspects of the deployment process to reduce the overhead required to maintain deployments.
+* We will maintain ongoing deployments of Aragon on Ropsten, Goerli, and Kovan testnests, similarly to how Rinkeby is supported by A1. For as long as the 1Hive sponsorship (AGP-90) remains funded at or above the current level.
```

Comment:
How do we get notified about updates to Aragon? I notice that there are sometimes pushed some minor updates that have no "media PR". Is this the AA's responsibility, or is A1 actually de facto in charge here, and if so, how do we establish a stable line of communication?

Reply:
Also in terms of maintaining this, do we require recurring funding to do so?

Reply:
> How do we get notified about updates to Aragon? I notice that there are sometimes pushed some minor updates that have no "media PR". Is this the AA's responsibility, or is A1 actually de facto in charge here, and if so, how do we establish a stable line of communication?

I think we can watch https://github.com/aragon/deployments/ 

> Also in terms of maintaining this, do we require recurring funding to do so?

My thinking is that we would have it mostly automated so once we have the process set up the ongoing costs would be minimal. So my assumption is that we would just use a small portion of the sponsorship to cover that overhead, if we determine that its more expensive than anticipated we can ask to have the sponsorship amount increased.

Reply:
My only concern is that that repository is actually used for something and that our deployments wouldn't be in that repository. I think the CLI _might_ use it for something? @0xGabi

Reply:
> My only concern is that that repository is actually used for something and that our deployments wouldn't be in that repository. I think the CLI _might_ use it for something? @0xGabi

Not sure I follow, I know that all of A1's deployments get registered there and any ipfs content that is there gets automatically pinned. 

It may not be the best process and we could propose changes if necessary, but it seems like it's currently the best way for us to monitor even minor deployments.

Reply:
Yeah, but I meant that it might be beneficial for our deployments to Kovan/Görli etc. to also be in that repository.

Reply:
> might be beneficial for our deployments to Kovan/Görli etc 

We could submit PRs for this, or request merge permission.